pkg/gadgets/run/tracer/run.go

```diff
@@ -104,6 +105,12 @@ func (g *GadgetDesc) ParamDescs() params.ParamDescs {
 			},
 			TypeHint: params.TypeString,
 		},
+		{
+			Key:         pullSecret,
+			Title:       "Pull secret",
+			Description: "Use the default (gadget-pull-secret) pull secret for the registry",
+			TypeHint:    params.TypeBool,
+		},
```

Comment:
IMO it should be a string and let the user define any secret they want. The default secret is useful to avoid having to repeat it each time, but if users want to use a different secret for a specific gadget, we should support it too.

Reply:
btw, sorry if I'm asking too much, but I think we're almost there to have support for both modes, so why not to implement them now?

Reply:
All good, I had my doubts about this already :D I agree we should allow using any custom secrets with `run` as it makes sense to specify any exiting secret at runtime. Remembering secret name across `deploy` can get tricky so makes sense to use a default secret.